### PR TITLE
Added more customization in ChannelViewStyle

### PR DESCRIPTION
**Pull request by buh**
Changes:
- opened `ChannelTableViewCell`.
- added `SeparatorStyle`.
- added customization for channel avatar with `AvatarViewStyle`.
- added a name number of lines.
- added a message number of lines.
- added a custom height for the channel cell.
- added spacing.
- added edge insets.

### CODE REVIEW

File: Sources/UI/Channels/ChannelTableViewCell.swift [hunk 1]
```diff
@@ -10,10 +10,21 @@ import UIKit
...
+    
+    /// A channel style.
+    public var style: ChannelViewStyle = ChannelViewStyle()
+    /// Checks if needds setup layout.
+    public private(set) var needsToSetup = true
     
     /// An avatar view.
-    public let avatarView: AvatarView = AvatarView(cornerRadius: 20)
+    public private(set) lazy var avatarView: AvatarView = {
+        if let avatarStyle = style.avatarViewStyle {
```
Comment (b-onc) [inline Sources/UI/Channels/ChannelTableViewCell.swift]: Why not just `return AvatarView(cornerRadius: style.avatarViewStyle?.radius ?? .channelAvatarRadius)`?

File: Sources/UI/Channels/ChannelTableViewCell.swift [hunk 2]
```diff
@@ -86,48 +94,68 @@ public final class ChannelTableViewCell: UITableViewCell, Reusable {
...
+            }
         }
         
         nameLabel.snp.makeConstraints { make in
-            make.left.equalTo(avatarView.snp.right).offset(CGFloat.messageInnerPadding)
-            make.right.lessThanOrEqualTo(dateLabel.snp.left).offset(-CGFloat.messageInnerPadding)
-            make.bottom.equalTo(avatarView.snp.centerY)
+            if hasAvatar {
+                make.bottom.equalTo(avatarView.snp.centerY).offset(style.spacing.vertical / -2).priority(999)
+                make.left.equalTo(avatarView.snp.right).offset(style.spacing.horizontal)
```
Comment (b-onc) [inline Sources/UI/Channels/ChannelTableViewCell.swift]: Not for this PR, but we should start using `leading` and `trailing` for labels, so in the future we can support right to left languages easily.

File: Sources/UI/Style/AvatarViewStyle.swift [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+/// Avatars style.
+public struct AvatarViewStyle: Hashable {
+    /// An avatar radius.
+    public let radius: CGFloat
+    /// A placeholder font.
+    public let placeholderFont: UIFont?
+    
+    /// A double value of `radius`.
+    public var size: CGFloat {
```
Comment (b-onc) [inline Sources/UI/Style/AvatarViewStyle.swift]: This is confusing: a circle's size is not double it's radius, so what is this value? Edge length?

File: Sources/UI/Channels/ChannelTableViewCell.swift (hunk 1 above)
Comment (buh) [reply Sources/UI/Channels/ChannelTableViewCell.swift]: Yes. I'll fix

File: Sources/UI/Channels/ChannelTableViewCell.swift (hunk 2 above)
Comment (buh) [reply Sources/UI/Channels/ChannelTableViewCell.swift]: A good point.

File: Sources/UI/Style/AvatarViewStyle.swift (hunk 3 above)
Comment (buh) [reply Sources/UI/Style/AvatarViewStyle.swift]: Avatars are rounded with some radius. Size is for width and height of it.

Comment (b-onc) [reply Sources/UI/Style/AvatarViewStyle.swift]: Can we rename it so it's clearer? Something like `squareEdge` or `squareEdgeLength` or `boundingSquareEdge`?

Comment (buh) [reply Sources/UI/Style/AvatarViewStyle.swift]: It's more than clear :) It's using a lot in other projects. It's actually diameter, but mostly id named as just size.

**Pull request merged**